### ISSUE TITLE
about class CurricularFace(nn.Module)

### ISSUE THREAD

**Issue by zhushumin** [HuangYG123/CurricularFace#14]
In this loss, you choose the hard_example by "mask = cos_theta > cos_theta_m; " and " hard_example = cos_theta[mask]". However the cos_theta is always larger than cos_thata_m. This code is useless and it does not match your paper. 

if   cos(θyi + m) ≥ cosθj then 
   N(t,cosθj) = cosθj; 
else 
   N(t,cosθj) = (t(k) + cosθj)cosθj ;

**Comment by HuangYG123** [comment]
cos_theta is the cosine similarity, and the shape is BxC. 
cos_theta_m is the target logit, and the shape is Bx1. 
The shape of mask is BxC. 
Why cos_theta is always larger than cos_thata_m?

**Comment by zhushumin** [comment]
Sorry, that's my fault. I miss understand your code.
